modernize(tags): bulk delete tags api [#32970]

hassandotcms opened the pull request:
### Proposed Changes
- Add deleteTags(String... tagIds) method to TagAPI interface
- Implement batch deletion using DotConnect.executeBatch() for performance
- Update REST endpoint to accept JSON body with tagIds array instead of path parameter
- Add comprehensive Postman tests for bulk delete

resolves: #32970 

This PR fixes: #32970

hassandotcms committed:
modernize(tags): add bulk delete API endpoint for tags

  - Add deleteTags(String... tagIds) method to TagAPI interface
  - Implement batch deletion using DotConnect.executeBatch() for performance
  - Update REST endpoint to accept JSON body with tagIds array instead of path parameter
  - Add comprehensive Postman tests for bulk delete and error cases

ref: #32970

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32970-bulk-delete-tags-api

fabrizzio-dotCMS commented on dotCMS/src/main/java/com/dotmarketing/tag/business/TagFactoryImpl.java:
```diff
@@ -446,6 +446,39 @@ public void deleteTag ( Tag tag ) throws DotDataException {
     }
 
+    @Override
+    public void deleteTagsInBatch(java.util.Collection<String> tagIds) throws DotDataException {
```
please add a test that covers this code on TagAPITest
test public void deleteTags(String... tagIds) throws DotDataException on the API

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32970-bulk-delete-tags-api

hassandotcms committed:
modernize(tags): add bulk delete tags api (#32970)

add integration test for bulk delete tags api. and fix failing postman test.

fabrizzio-dotCMS committed:
Merge branch 'main' into 32970-bulk-delete-tags-api

hassandotcms committed:
Merge branch 'main' into 32970-bulk-delete-tags-api

hassandotcms committed:
Merge branch 'main' into 32970-bulk-delete-tags-api

hassandotcms committed:
Merge branch 'main' into 32970-bulk-delete-tags-api

wezell reviewed (changes requested):
Guys, I removed from queue because I wanted to talk about permissions on this endpoint - unless I am mistaken, deleting tags is very much like changing content, especially as some of our tags have business meaning.  It does not look like this API  requires any permissions to do a bulk delete of tags.

hassandotcms committed:
modernize(tags): rest api to delete tags in bulk #32970

- Permission checks added: Before deleting a tag, verifies user has EDIT permission on ALL associated contentlets; orphan tags can be deleted freely
- Bulk delete returns BulkResultView: V2 delete endpoint now returns success count and per-tag failure details instead of boolean

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32970-bulk-delete-tags-api

hassandotcms committed:
moderniz(tags): bulk delete tags #32970
- check admin permission check to delete tags.

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32970-bulk-delete-tags-api

wezell commented on dotCMS/src/main/java/com/dotmarketing/tag/business/TagAPIImpl.java:
```diff
@@ -422,6 +426,93 @@ public void deleteTag ( String tagId ) throws DotDataException {
+     * @return null if deletion is allowed, or an error message if denied
+     * @throws DotDataException if there's a data access error
+     */
+    private String checkTagDeletePermission(final String tagId, final User user) throws DotDataException {
```
Also this method should either return a `boolean` or void.  I would rather a boolean as we should not use exceptions for flow control.

wezell reviewed (changes requested):
Not bad but added some comments

hassandotcms committed:
modernize(permissions): bulk delete tag rest api #32970

 1. Simplified API signature & behavior: Changed Map<String, Object> to List<String> for bulk delete request, made non-existent tags skip silently (idempotent), and merged checkTagDeletePermission() into
  canDeleteTag()
  2. Removed exception-based flow control: Changed canDeleteTag() and deleteTag(User, String) to return boolean instead of throwing DotSecurityException

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v2/tags/TagResource.java:
```diff
@@ -579,39 +582,110 @@ public ResponseEntityRestTagView getTagsByNameOrId(@Context final HttpServletReq
+                                                          content = @Content(schema = @Schema(type = "object",
+                                                                                            description = "Object containing array of tag IDs",
+                                                                                            example = "{\"tagIds\": [\"tag-123\", \"tag-456\", \"tag-789\"]}")))
+                                               final java.util.Map<String, Object> deleteRequest) {
```
updated.

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v2/tags/TagResource.java:
```diff
@@ -579,39 +582,110 @@ public ResponseEntityRestTagView getTagsByNameOrId(@Context final HttpServletReq
+                // Check if tag exists
+                final Tag tag = tagAPI.getTagByTagId(tagId);
+                if (tag == null || !UtilMethods.isSet(tag.getTagId())) {
+                    failedToDelete.add(new FailedResultView(tagId, "Tag does not exist"));
```
makes sense, fixed.

wezell reviewed (approved):
Looks good after changes.

hassandotcms committed:
Merge branch 'main' into 32970-bulk-delete-tags-api
